extensions/amp-analytics/0.1/events.js

```diff
@@ -244,18 +269,31 @@ export class SignalTracker extends EventTracker {
                 selectionMethod),
             `Element "${selector}" not found`);
         target = element;
-        return element.signals();
+        return this.getElementSignal(element, eventType);
       });
     }
 
     // Wait for the target and the event signal.
-    signalsPromise.then(signals => signals.whenSignal(eventType)).then(() => {
+    signalsPromise.then(() => {
       listener(new AnalyticsEvent(target, eventType));
     });
     return NO_UNLISTEN;
   }
-}
 
+  /**
+   * @override
+   */
+  getRootSignal(eventType) {
+    return this.root.signals().whenSignal(eventType);
+  }
+
+  /**
+   * @override
+   */
+  getElementSignal(element, eventType) {
+    return element.signals().whenSignal(eventType);
```

Comment:
Since `element` is defined as `Element` and not `AmpElement` (which is correct), please check for `typeof element.signals == 'function'`.